python/nano/src/bigdl/nano/deps/ipex/ipex_inference_bf16_model.py

```diff
@@ -38,13 +48,90 @@ def __init__(self, model, input_sample=None, use_ipex=False,
                the parameter will be ignored if use_ipex is False.
         :param from_load: this will only be set by _load method.
         '''
+        invalidInputError(
+            self._has_bf16_isa or self._allow_non_bf16,
+            errMsg="Your machine or OS doesn't support BF16 instructions."
+        )
+        dtype = torch.bfloat16 if self._has_bf16_isa else None
         PytorchIPEXJITModel.__init__(self, model, input_sample=input_sample, use_ipex=use_ipex,
-                                     dtype=torch.bfloat16, use_jit=use_jit,
+                                     dtype=dtype, use_jit=use_jit,
                                      channels_last=channels_last, from_load=from_load)
 
-    @autocast()
+    @property
+    def _allow_non_bf16(self):
+        """
+        ALLOW_NON_BF16_ISA indicates if we restrict bf16 instructions support to be available.
+        ALLOW_NON_BF16_ISA='1' sometimes helps debug and test cases without AVX512 or AMX
+
+        :return: The bool value of ALLOW_NON_BF16_ISA
+        """
+        return os.environ.get("ALLOW_NON_BF16_ISA", None) == '1'
```

Comment:
I think this is not necessary? If running on non-avx512 machine, it's better to raise error.